dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/ArgumentTypesResolver.java

```diff
@@ -126,38 +101,24 @@ private String planSerDeMethodName(TypeName in) {
     }
 
     private boolean isRawRequestBody(TypeMirror in) {
-        return types.isSameType(in, requestBodyType);
+        return context.isSameTypes(in, RequestBody.class);
     }
 
     private Optional<OptionalType> getOptionalType(Element paramContext, TypeMirror typeName) {
-        // "They tryin'a make me use a visitor, but I say: NO, NO NO!"
-        if (!(typeName instanceof DeclaredType)) {
-            return Optional.empty();
-        }
-        DeclaredType declaredType = (DeclaredType) typeName;
-        if (types.isSameType(declaredType, optionalIntType)) {
+        if (context.isSameTypes(typeName, OptionalInt.class)) {
```

Comment:
OptionalDouble + OptionalLong?

Reply:
I'll add those in a separate PR, the serializer and header only works for these 2 as well...